eclipse-collections/src/main/java/org/eclipse/collections/impl/map/immutable/ImmutableDoubletonMap.java

```diff
@@ -102,9 +101,9 @@ public V get(Object key)
     }
 
     @Override
-    public Set<K> keySet()
+    public ImmutableKeySet<K> keySet()
     {
-        return Sets.immutable.with(this.key1, this.key2).castToSet();
+        return new ImmutableKeySet<>(Sets.immutable.with(this.key1, this.key2));
```

Comment:
@johnjstuart what's the point of wrapping an already `ImmutableSet` to an `ImmutableSet`? (here and all the other subclasses of `AbstractImmutableMap`? Just return it.

Reply:
@johnjstuart but `ImmutableDoubletonMap` implements `ImmutableSet`, and `ImmutableSet.keySet()` is meant to return an `ImmutableSet`, not a Set. Note also that all `ImmutableSet` implementations already implements `Set`, see https://github.com/eclipse/eclipse-collections/blob/master/eclipse-collections-api/src/main/java/org/eclipse/collections/api/set/ImmutableSet.java#L42

Reply:
@johnjstuart hence I think it is useless to do all this wrapping

Reply:
Yes, that is my understanding as well.

Reply:
Ok, now I understand better, thank you :) it's a bit strange to expose internal classes like this to the public API. Should we introduce a dedicated public interface that extends both `Set` and `ImmutableSet` for example to represent this instead of leaking a class meant to be internal?

Reply:
For example, `ImmutableSet.castToSet()` would return such interface instead of just returning a `Set`!